overlord/hookstate/ctlcmd/model.go

```diff
@@ -101,74 +99,120 @@ func (c *modelCommand) reportError(format string, a ...interface{}) {
 	w.Flush()
 }
 
-// checkGadgetOrModel verifies that the current snap context is either a gadget snap or that
-// the snap shares the same publisher as the current model assertion.
-func (c *modelCommand) checkGadgetOrModel(st *state.State, snapName string) error {
-	st.Lock()
-	defer st.Unlock()
+func interfaceConnected(st *state.State, snapName, ifName string) bool {
+	conns, err := ifacerepo.Get(st).Connected(snapName, ifName)
+	return err == nil && len(conns) > 0
+}
+
+// hasSnapdControl returns true if the requesting snap has the snapd-control plug
+// and only if it is connected as well.
+func hasSnapdControl(st *state.State, snapInfo *snap.Info) bool {
+	// Always get the current info even if the snap is currently
+	// being operated on or if its disabled.
+	if snapInfo.Broken != "" {
+		return false
+	}
 
+	// The snap must have a snap declaration (implies that
+	// its from the store)
+	if _, err := assertstate.SnapDeclaration(st, snapInfo.SideInfo.SnapID); err != nil {
+		return false
+	}
```

Comment:
we don't need to check this here, it's a check that we do only very rarely (for managed refreshs so far) as it blocks local development

Reply:
Good point, this was removed automatically as a part of matching the approach in refresh.go